predicate/logical.go

```diff
@@ -31,7 +31,7 @@ func (op LogicalOperator) Value() (datatypes.Node_Logical, error) {
 // LogicalNode is a node type includes a logical expression with other nodes.
 type LogicalNode struct {
 	Operator LogicalOperator `json:"operator"`
-	Children []Node          `json:"children"`
+	Children [2]Node         `json:"children"`
```

Comment:
What warranted this change?

Reply:
had same question, is this to avoid the panic on zero value slice(`nil`)?

Reply:
this line got me, didn't saw it in the previous commit
https://github.com/influxdata/influxdb/blob/68cfd386c377d1869dd54805caefdd0c08c4860b/tsdb/tsm1/predicate.go#L163